packages/@aws-cdk/aws-logs/lib/log-retention.ts

```diff
@@ -151,6 +159,29 @@ class LogRetentionFunction extends Construct implements cdk.ITaggable {
       // creates a CF circular dependency.
       resources: ['*'],
     }));
+    // if removalPolicy is DESTROY, add action for DeleteLogGroup and DeleteLogStream
+    if (props.removalPolicy === cdk.RemovalPolicy.DESTROY) {
+      role.addToPrincipalPolicy(new iam.PolicyStatement({
+        actions: ['logs:DeleteLogGroup'],
+        //Only allow deleting the specific log group.
+        resources: [cdk.Stack.of(this).formatArn({
+          service: 'logs',
+          resource: 'log-group',
+          resourceName: props.logGroupName+':*',
```

Comment:
```suggestion
          resourceName: props.logGroupName,
```

Reply:
After I removed the ':*', I manually deploy and destroy the stack in integ.log-retention.ts. However, the destroy failed, the error message is: the loggroup is not allowed to delete. After I added the ':*' again, the destroy succeeded. I found the arn of log group is 'arn:aws:logs:us-east-1:account-id:log-group:logRetentionLogGroup:*' from the  cloud watch
![2b1fbfa6bc14d14ad85faf66195dce0](https://user-images.githubusercontent.com/22937131/183458138-68be7e51-9f6b-49e1-8100-775e35ba9405.jpg)
![fd2d01cdbe057bb4fceff4eab18905e](https://user-images.githubusercontent.com/22937131/183458163-252ebc16-62b7-4987-a947-8b2ee727f842.jpg)

.

Reply:
OK. I was incorrect here, but let's at least us better string formatting. See my edited suggestion.